src/test/java/com/example/CustomerDaoTest.java

```diff
@@ -0,0 +1,145 @@
+package com.example;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.*;
+import static org.mockito.Mockito.when;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.nio.file.StandardOpenOption;
+import java.util.List;
+import javax.persistence.EntityManager;
+import javax.persistence.EntityManagerFactory;
+import javax.persistence.EntityTransaction;
+import javax.persistence.Persistence;
+import javax.persistence.criteria.CriteriaBuilder;
+import org.example.DAO.imp.CustomerDao;
+import org.example.model.Customer;
+import org.junit.jupiter.api.*;
+
+public class CustomerDaoTest {
+  private static final String LOG_PATH = "logs/app.log";
+
+  EntityManagerFactory etf;
+  CustomerDao customerDao;
+  Customer customer;
+
+  @BeforeEach
+  public void tearDown() throws IOException {
+    Files.write(
+        Paths.get(LOG_PATH),
+        new byte[0],
+        StandardOpenOption.TRUNCATE_EXISTING,
+        StandardOpenOption.CREATE);
+    etf = Persistence.createEntityManagerFactory("persistence");
+    customerDao = new CustomerDao(etf);
+    customer =
+        new Customer(null, "QiyanaTech", "test@gmail.com", "Rua Tiberius Dourado", "123456789");
+  }
+
+  @Test
```

Comment:
While this test works, if you really want to test if a customer was added, you need to try and `find`/`get` it afterward.

What are you really testing here, otherwise?

Hint: an `id` is populated by the database after any add/save operation.